### PR TITLE
Kotlin Refactor Configuration Classes

### PR DESCRIPTION
**Pull request by sshropshire**
### Summary of changes

 - This is an accumulation of all Kotlin changes made so far.
 - We added support for documentation generation so that we can continue the refactor in the main line of the repo.

 ### Checklist

 ~- [ ] Added a changelog entry~

- @sshropshire
- @jcnoriega

### CODE REVIEW

**Review by jaxdesmarais — APPROVED** [review]
👻

File: BraintreeCore/src/main/java/com/braintreepayments/api/AnalyticsConfiguration.kt [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+ * @property url url of the Braintree analytics service.
+ */
+internal data class AnalyticsConfiguration(
+    val url: String = ""
+) {
+
+    constructor(json: JSONObject?): this(
+        Json.optString(json, URL_KEY, "")
+    )
+    /**
```
Comment (scannillo) [inline BraintreeCore/src/main/java/com/braintreepayments/api/AnalyticsConfiguration.kt:17]:
```suggestion

    /**
```

File: BraintreeCore/src/main/java/com/braintreepayments/api/BraintreeApiConfiguration.kt [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+import org.json.JSONObject
+
+/**
+ * Contains the remote Braintree API Configuration for the Braintree SDK.
+ * @property accessToken The Access Token for Braintree API.
+ * @property url the base url for accessing Braintree API.
+ */
+internal data class BraintreeApiConfiguration(
+    val accessToken: String = "",
+    val url: String = ""
```
Comment (scannillo) [inline BraintreeCore/src/main/java/com/braintreepayments/api/BraintreeApiConfiguration.kt:13]: Why default to empty string, when the `optString()` is already falling back to empty string?

File: BraintreeCore/src/main/java/com/braintreepayments/api/GraphQLConfiguration.kt [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+internal data class GraphQLConfiguration(
+    val url: String = "",
+    private val features: Set<String> = HashSet<String>()
+) {
+
+    constructor(json: JSONObject?) : this(
+        Json.optString(json, GraphQLConstants.Keys.URL, ""),
+        parseJsonFeatures(json?.optJSONArray(GraphQLConstants.Keys.FEATURES))
+    )
+    /**
```
Comment (scannillo) [inline BraintreeCore/src/main/java/com/braintreepayments/api/GraphQLConfiguration.kt:22]:
```suggestion

    /**
```

**Review by scannillo — APPROVED** [review]
Kotlin is kool! 🎸

File: BraintreeCore/src/main/java/com/braintreepayments/api/BraintreeApiConfiguration.kt (hunk 2 above)
Comment (sshropshire) [reply BraintreeCore/src/main/java/com/braintreepayments/api/BraintreeApiConfiguration.kt:13]: That's a good call. I'll remove these default assignments.

**Pull request merged**